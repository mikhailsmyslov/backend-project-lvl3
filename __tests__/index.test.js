import nock from 'nock';
import os from 'os';
import path from 'path';
import fs from 'fs';
import cheerio from 'cheerio';
import pageLoader from '../src';

nock.disableNetConnect();

const baseUrl = 'https://google.com';
const repliesPath = `${__dirname}/__fixtures__/replies`;
const referenceHtml = fs.readFileSync(`${__dirname}/__fixtures__/expected/google-com.html`, 'utf-8');
const expectedHtml = cheerio.load(referenceHtml).html();

nock(baseUrl)
  .get('/')
  .replyWithFile(200, `${repliesPath}/google.html`);
nock(baseUrl)
  .get('/images/branding/googlelogo/2x/googlelogo_color_272x92dp.png')
  .replyWithFile(200, `${repliesPath}/googlelogo_color_272x92dp.png`);
nock(baseUrl)
  .get('/tia/tia.png')
  .replyWithFile(200, `${repliesPath}/tia.png`);

let tmpDirPath = '';

beforeEach(async () => {
  tmpDirPath = await fs.promises.mkdtemp(path.join(os.tmpdir(), 'pageloadertest-'));
});

test('should work', async () => {
  await pageLoader(baseUrl, tmpDirPath);
  const html = await fs.promises.readFile(`${tmpDirPath}/google-com.html`, 'utf-8');
  expect(html).toBe(expectedHtml);
});
