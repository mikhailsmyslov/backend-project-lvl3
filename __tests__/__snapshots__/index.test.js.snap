// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FS error 1`] = `"ENOENT: no such file or directory, mkdir '/notExist/testpage-com_files'"`;

exports[`HTTP code is not 200 1`] = `"Expected HTTP status code 200, but recieved 204"`;

exports[`HTTP error 4xx 1`] = `"Request failed with status code 404"`;
